Code/Engine/GameEngine/StateMachine/Implementation/StateMachine.cpp

```diff
@@ -350,6 +350,9 @@ ezResult ezStateMachineInstance::SetState(ezStateMachineState* pState)
     return SetState(pState->GetNameHashed());
   }
 
+  if (m_pCurrentState == pState)
+    return EZ_SUCCESS;
```

Comment:
Why is this not moved to the start of the function? From the looks of it, we will always execute `SetState` no matter what, wo it's not a real early out? I also don't quite get this function, it seems to be doing most of the stuff that `SetStateInternal` does, but only if `pState == nullptr` in which case, should we be doing anything at all? Or if `m_pDescription == nullptr` in which case, shouldn't we fail the function immediately instead of returning success?